.github/workflows/ruby.yml

```diff
@@ -0,0 +1,63 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      DATABASE_URL: mysql2://root:@127.0.0.1:3306/wca_test
+      RACK_ENV: test
+    defaults:
+      run:
+        working-directory: WcaOnRails
+    steps:
+      - uses: actions/checkout@v2
+      - uses: fregante/setup-git-user@v1 # set up dummy user.name and user.email in git so that Overcommit doesn't explode
+      - name: Activate MySQL # as per https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#mysql
+        run: |
+          sudo systemctl start mysql.service
+          mysql -e "ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';" -u root -proot
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          working-directory: WcaOnRails
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+      - uses: actions/setup-node@v2 # this also includes Yarn
+        with:
+          node-version: '12'
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}
+          # restore-keys: |
+          #   ${{ runner.os }}-node_modules
```

Comment:
Nit: leftover

Reply:
Not necessarily a leftover. `restore-keys` is pretty much a backup mechanism: If it doesn't find a cache stored under the key provided by `key`, it tries `restore-keys` in order as fallback. I left this in to debate whether we want to use fallbacks here, i.e. rely on older caches where the hash of `yarn.lock` doesn't match. Opinions?

Reply:
Ah, yeah I think using an older cache should be fine, especially given the automated dependency updates we get.